rules/ruleset.go

```diff
@@ -229,12 +366,18 @@ func (as *activeRuleSet) toRollupResults(id []byte, cutoverNanos int64, targets
 				continue
 			}
 			if res > 0 {
-				targetTagIdx++
-				continue
+				break
 			}
 			hasMoreTags = tagIter.Next()
 		}
 		tagIter.Close()
+		// If not all the target tags are found in the id, this is considered
+		// an ineligible rollup target. In practice, this should never happen
+		// because the UI requires the list of rollup tags should be a subset
+		// of the tags in the metric selection filter.
+		if targetTagIdx < len(target.Tags) {
+			continue
```

Comment:
I'd prefer not, the assumption we made in other components shouldn't affect the logic here, plus other users may make different assumptions.